tests/src/androidTest/java/com/squareup/duktape/DuktapeGetTest.java

```
@@ -388,4 +388,189 @@
+    Object[] sorted = sorter.sort(original);
+    assertThat(sorted).isEqualTo(new Object[]{1.0, 2.0, 3.0, 4.0});
+    assertThat(original).isNotSameAs(sorted);
+
+    assertThat(sorter.sort(new Object[]{"b", "d", null, "a"}))
+        .isEqualTo(new String[]{"a", "b", "d", null});
+  }
+
+  interface StringSorter {
+    String[] sort(String[] args);
+  }
+
+  @Test public void arraysOfStrings() {
+    duktape.evaluate(SORTER_FUNCTOR);
+
+    StringSorter sorter = duktape.get("Sorter", StringSorter.class);
+
+    assertThat(sorter.sort(new String[]{ "b", "d", "c", "a" }))
+        .isEqualTo(new String[]{ "a", "b", "c", "d" });
+
+    assertThat(sorter.sort(new String[]{"b", "d", null, "a"}))
+        .isEqualTo(new String[]{"a", "b", "d", null});
+
+    // Replace the sorter with something broken.
+    duktape.evaluate(""
+        + "Sorter.sort = function(v) {"
+        + "  return [ 'a', 'b', 3, 'd' ];\n"
+        + "};");
+    try {
+      sorter.sort(new String[]{ "b", "d", "c", "a" });
+      fail();
+    } catch (IllegalArgumentException expected) {
+      assertThat(expected).hasMessage("Cannot convert return value 3 to String");
+    }
+  }
+
+  interface DoubleSorter {
+    double[] sort(double[] args);
+  }
+
+  @Test public void arraysOfDoubles() {
+    duktape.evaluate(SORTER_FUNCTOR);
+
+    DoubleSorter sorter = duktape.get("Sorter", DoubleSorter.class);
+
+    assertThat(sorter.sort(new double[] { 2.9, 2.3, 3, 1}))
+        .isEqualTo(new double[]{ 1.0, 2.3, 2.9, 3.0 }, 0.0);
+
+    // Replace the sorter with something broken.
+    duktape.evaluate(""
+        + "Sorter.sort = function(v) {"
+        + "  return [ 1, 2, null, 4 ];\n"
+        + "};");
+    try {
+      sorter.sort(new double[0]);
+      fail();
+    } catch (IllegalArgumentException expected) {
+      assertThat(expected).hasMessage("Cannot convert return value null to double");
+    }
+  }
+
+  // Note, we return double[] and Double[] since we can't return ints from JavaScript.
```

This is the same as the scalar case - we forbid int as a return type.

We totally could support int[] (and int), but we would end up silently truncating floating point numbers when the return type is declared int - same as a the scalar case.